x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -0,0 +1,495 @@
+    ]);
+
+    if (!dataView) {
+      return null;
+    }
+
+    return (
+      <StatefulEventContext.Provider value={activeStatefulEventContext}>
+        <StyledTimelineUnifiedDataTable>
+          {(dataLoadingState === DataLoadingState.loading ||
+            dataLoadingState === DataLoadingState.loadingMore) && (
+            <StyledEuiProgress data-test-subj="discoverDataGridUpdating" size="xs" color="accent" />
+          )}
+          <DataGridMemoized
+            ariaLabelledBy="timelineDocumentsAriaLabel"
+            className={'udtTimeline'}
+            columns={defaultColumnIds}
+            expandedDoc={expandedDoc}
+            dataView={dataView}
+            showColumnTokens={true}
+            loadingState={dataLoadingState}
+            onFilter={onAddFilter as DocViewFilterFn}
+            onResize={onResizeDataGrid}
+            onSetColumns={onSetColumns}
+            onSort={!isTextBasedQuery ? onSort : undefined}
+            rows={tableRows}
+            sampleSizeState={sampleSize || 500}
+            onUpdateSampleSize={onUpdateSampleSize}
+            setExpandedDoc={onSetExpandedDoc}
+            settings={tableSettings}
+            showTimeCol={showTimeCol}
+            isSortEnabled={true}
+            sort={sortingColumns}
+            rowHeightState={3}
+            isPlainRecord={isTextBasedQuery}
+            rowsPerPageState={itemsPerPage}
+            onUpdateRowsPerPage={onChangeItemsPerPage}
+            onUpdateRowHeight={onUpdateRowHeight}
+            onFieldEdited={onFieldEdited}
+            cellActionsTriggerId={SecurityCellActionsTrigger.DEFAULT}
+            services={services}
+            visibleCellActions={3}
+            externalCustomRenderers={customColumnRenderers}
+            renderDocumentView={() => <></>}
+            rowsPerPageOptions={itemsPerPageOptions}
+            showFullScreenButton={false}
+            useNewFieldsApi={true}
+            maxDocFieldsDisplayed={50}
+            consumer="timeline"
+            totalHits={totalCount}
+            onFetchMoreRecords={handleFetchMoreRecords}
+            configRowHeight={3}
+            showMultiFields={true}
+            cellActionsMetadata={cellActionsMetadata}
+            externalAdditionalControls={additionalControls}
+          />
+          {showExpandedDetails && (
+            <DetailsPanel
+              browserFields={browserFields}
+              handleOnPanelClosed={handleOnPanelClosed}
+              runtimeMappings={runtimeMappings}
+              tabType={activeTab}
+              scopeId={timelineId}
+              isFlyoutView
+            />
+          )}
+        </StyledTimelineUnifiedDataTable>
+      </StatefulEventContext.Provider>
+    );
+  }
+);
+
+export const TimelineDataTable: React.FC<CommonDataTableProps> = memo((props) => {
+  const dataView = useGetScopedSourcererDataView({ sourcererScope: SourcererScopeName.timeline });
+  if (!dataView) {
+    throw new Error('DataView must be defined');
```

Comment:
Ahh.. maybe we do something here to handle this situation. Render something that highlights the dataView is missing?